core/src/test/java/org/eclipse/hono/util/MessageHelperTest.java

```diff
@@ -325,6 +341,64 @@ public void testNewMessageUsesGivenTtlValue() {
         assertThat(message.getTtl()).isEqualTo(timeToLive.toMillis());
     }
 
+    /**
+     * Verifies that the content-type of a newly created event message
+     * is set to the "application/vnd.eclipse-hono-empty-notification" content-type
+     * even if the payload is {@code null}.
```

Comment:
this sounds like the content-type is set *despite* the fact that the body is null. However, my understanding is that this should happen *only* if the body is null, right? So, FMPOV this should read
> ... is set to the "application/vnd.eclipse-hono-empty-notification" content-type if the payload is null

Reply:
> this should happen only if the body is null, right?

From an overall perspective yes, but this condition is currently not enforced in the MessageHelper methods (there is also no automatic setting of "application/vnd.eclipse-hono-empty-notification" if the payload is null). 
(Throwing an IllegalArgumentException for "eclipse-hono-empty-notification" with payload != null (or not empty) would be the method doing too much, I think.)

So it's more like this content-type gets applied *even though* the payload is null, because normally the content-type parameter isn't applied if the payload is null. There's only the special case of the "application/vnd.eclipse-hono-empty-notification" content type for which this is done.

Anyway, I've changed the text now to:
> Verifies that the "application/vnd.eclipse-hono-empty-notification" content-type
can be set on a newly created event message with a `null` payload.